File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // Create an unordered set to store unique elements
        unordered_set<int> numSet(nums.begin(), nums.end());

        int longestStreak = 0;

        // Iterate through each number in the set
        for (int num: numSet){
            // Check if num is the start of a new sequence
            if(numSet.find(num - 1) == numSet.end()){ //'num - 1' not in the sequence
                int currentNum = num;
                int currentStreak = 1;

                // Count the length of the current sequence
                while(numSet.find(currentNum + 1) != numSet.end()){
                    currentNum++;
                    currentStreak++;
                }

                // Update the longest streak
                longestStreak = max(longestStreak, currentStreak);
            }
        }
        return longestStreak;
    }
};

File: Data Structures & Algorithms/subsets/submission-3.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []

        def backtrack(start, current):
            res.append(current[:])

            for i in range(start, len(nums)):
                current.append(nums[i])

                backtrack(i + 1, current)

                current.pop()
        

        backtrack(0, [])

        return res